src/utils/maybeExecuteLifecycleScript.ts

```diff
@@ -0,0 +1,24 @@
+import { Workspace, structUtils } from '@yarnpkg/core'
+import { execute } from '@yarnpkg/shell'
+
+import logger from '../logging'
+
+export default async function maybeExecuteLifecycleScript(
+    workspace: Workspace,
+    scriptIdentifier: string,
```

Comment:
(nb)
type LifecycleScriptName = 'prepare' | 'prepack' | 'postpack' // etc...

```suggestion
    scriptIdentifier: LifecycleScriptName,
```